internal/controller/controllers/agentserviceconfig_controller_storage_validation_test.go

```diff
@@ -0,0 +1,431 @@
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	clnt "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Agent service config controller storage validation", func() {
+	var (
+		ctx        context.Context
+		cluster    *testing.FakeCluster
+		client     clnt.Client
+		subject    *aiv1beta1.AgentServiceConfig
+		key        clnt.ObjectKey
+		request    ctrl.Request
+		reconciler *AgentServiceConfigReconciler
+	)
+
+	BeforeEach(func() {
+		// Create a context for the test:
+		ctx = context.Background()
+
+		// Create the fake cluster:
+		cluster = testing.NewFakeCluster().
+			Logger(logger).
+			Build()
+
+		// Create the client:
+		client = cluster.Client()
+
+		// Create the default object to be reconciled, with a storage configuration that
+		// passes all validations. Specific tests will later adapt these settings as needed.
+		subject = &aiv1beta1.AgentServiceConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "agent",
+			},
+			Spec: aiv1beta1.AgentServiceConfigSpec{
+				DatabaseStorage: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Ti"),
+						},
+					},
+				},
+				FileSystemStorage: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Ti"),
+						},
+					},
+				},
+				ImageStorage: &corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Ti"),
+						},
+					},
+				},
+			},
+		}
+		key = clnt.ObjectKeyFromObject(subject)
+		request = ctrl.Request{
+			NamespacedName: types.NamespacedName{
+				Name: subject.Name,
+			},
+		}
+
+		// Create the reconciler:
+		reconciler = &AgentServiceConfigReconciler{
+			AgentServiceConfigReconcileContext: AgentServiceConfigReconcileContext{
+				Client:   client,
+				Scheme:   cluster.Scheme(),
+				Log:      logrusLogger,
+				Recorder: cluster.Recorder(),
+			},
+			Namespace: "assisted-installer",
+		}
+	})
+
+	AfterEach(func() {
+		cluster.Stop()
+	})
+
+	When("Objects haven't been created", func() {
+		It("Rejects small database storage", func() {
+			subject.Spec.DatabaseStorage.Resources.Requests = corev1.ResourceList{
+				corev1.ResourceStorage: resource.MustParse("1Mi"),
+			}
+			err := client.Create(ctx, subject)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func(g Gomega) {
+				_, err := reconciler.Reconcile(ctx, request)
+				g.Expect(err).ToNot(HaveOccurred())
```

Comment:
When you don't care about the value returned, then gomega Expect looks only at the last parameter in the return value, so you can write more concise:

```suggestion
				g.Expect(reconciler.Reconcile(ctx, request)).To(Succeed())
```

Reply:
(this can be used in many other places in your code, not just this line)

Reply:
I am aware of that, but I prefer to do it this way because I find nested function calls harder to read. Probably just a matter of taste. I will change it if you insist :-) .

Reply:
Nah I don't insist, just a personal preference. But don't do it if you don't prefer it as well

Reply:
I will leave it as it is then, thanks @omertuc. Note that I meant it: if you had insisted I would happily have changed it.